lib/client/connection.js

```diff
@@ -476,6 +478,12 @@ Connection.prototype.send = function(message) {
   this.socket.send(JSON.stringify(message));
 };
 
+Connection.prototype.ping = function() {
```

Comment:
From discussion today - let's throw an error if the connection state is not connected.

This is to guard against a bunch of ping requests queueing up at the socket layer if the socket is currently disconnected or connecting, and having them suddenly flood the socket when eventually connected.